core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -155,7 +155,7 @@ object ExternalStoreUtils {
         // Already connected to SnappyData in embedded mode.
         Constant.DEFAULT_EMBEDDED_URL + ";host-data=false;mcast-port=0"
       case SplitClusterMode(_, _) =>
-        ServiceUtils.getLocatorJDBCURL(sc) + "/route-query=false"
+        ServiceUtils.getLocatorJDBCURL(sc) + ";route-query=false"
```

Comment:
The change is only applicable for split cluster mode. In case of secondary locators the url was not formed properly and was causing the issue.